scripts/showcase/samplesheet-prep-test.py

```diff
@@ -1,55 +1,399 @@
-    # Select valid samples (WGS and WTS currently) from tracking sheet and use a query function on samplesheet to select on the match in trackingsheet
-    valid_samples = trackingsheet.query('Type==@stype')['Sample_ID (SampleSheet)'].unique().tolist()
-    samplesheet_tmp = samplesheet.query("Sample_ID in @valid_samples")
-    #assign same header as 'samplesheet_metadata' df to 'samplesheet_tmp'
-    col_rename=0
-    for col in samplesheet_tmp:
-        samplesheet_tmp.rename(columns={col:col_rename}, inplace=True)
-        col_rename=col_rename+1
-    #concatenate sample sheet metadata and modified samplesheeta. Involves a hack to a
-    samplesheet_write=pd.concat([samplesheet_metadata, samplesheet_header, samplesheet_tmp], axis=0, ignore_index=True)
-    #write output to file
-    samplesheet_name = stype.join(["SampleSheet_", ".csv"])
-    samplesheet_write.to_csv(os.path.join(args.outputPath, samplesheet_name), header=False, index=False)
\ No newline at end of file
+# Globals
+HEADER_LINE_PRECURSOR = "[Data]"
+V2_COLUMN_CHANGES = {'I7_Index_ID': 'I7_Index', 'I5_Index_ID': 'I5_Index'}
+V2_METADATA_COLUMN_CHANGES = {"Adapter": "AdapterRead1"}
+VALID_SAMPLE_TYPES = {"TSO500": ["TSO"],
+                      "V2": ["WGS", "WTS"]}
+OMITTED_YEAR_SHEETS = ["2018"]
+
+# Set logs
+LOGGER_STYLE = '%(asctime)s - %(levelname)-8s - %(funcName)-20s - %(message)s'
+CONSOLE_LOGGER_STYLE = '%(funcName)-12s: %(levelname)-8s %(message)s'
+LOGGER_DATEFMT = '%y-%m-%d %H:%M:%S'
+THIS_SCRIPT_NAME = "SAMPLESHEET TESTER"
+
+
+def initialise_logger():
+    """
+    Return the logger in a nice logging format
+    :return:
+    """
+    # Initialise logger
+    # set up logging to file - see previous section for more details
+    logging.basicConfig(level=logging.DEBUG,
+                        format=LOGGER_STYLE,
+                        datefmt=LOGGER_DATEFMT)
+    # define a Handler which writes INFO messages or higher to the sys.stderr
+    console = logging.StreamHandler()
+    console.setLevel(logging.INFO)
+    # set a format which is simpler for console use
+    formatter = logging.Formatter(CONSOLE_LOGGER_STYLE)
+    # tell the handler to use this format
+    console.setFormatter(formatter)
+    # add the handler to the root logger
+    #logging.getLogger('').addHandler(console)
+
+
+def get_logger():
+    """
+    Get the name of where this function was called from - return a logging object
+    Use a trackback from the inspect to do this
+    :return:
+    """
+    return logging.getLogger(THIS_SCRIPT_NAME)
+
+
+# Get logger
+initialise_logger()
+logger = get_logger()
+
+
+def get_args():
+    """
+    Read args
+    return: argparse dict
+    """
+    # Initialise parser
+    parser = argparse.ArgumentParser()
+    # Place arguments
+    parser.add_argument("--samplesheet", "-s", type=str, required=True)
+    parser.add_argument("--trackingsheet", "-t", type=str, required=True)
+    parser.add_argument("--outputFile", "-o", type=str, required=True)
```

Comment:
update argument change to outputpath - as we are now writing multiple samplesheets

Reply:
https://github.com/umccr/infrastructure/pull/16/files#diff-9803ecbbdf15c05af6e26ccc75f88c84R72-R74

Reply:
`--outputFile` now changed to `--outputPath`.